packages/eslint-plugin/rules/i18n-hyphenated-range.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	getTextContentFromNode,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+} = require( '../utils' );
+
+const EN_DASH = '–';
+const HYPHEN_IN_RANGE = /(\d\s+-\s+\d)|(\d-\d)/g;
+
+function replaceHyphenWithEnDash( string ) {
+	if ( string.match( HYPHEN_IN_RANGE ) ) {
+		return string.replace( '-', EN_DASH );
+	}
+}
+
+// see eslint-plugin-wpcalypso.
+function makeFixerFunction( arg ) {
+	return ( fixer ) => {
+		switch ( arg.type ) {
+			case 'TemplateLiteral':
+				return arg.quasis.reduce( ( fixes, quasi ) => {
+					if (
+						'TemplateElement' === quasi.type &&
+						quasi.value.raw.match( HYPHEN_IN_RANGE )
+					) {
+						fixes.push(
+							fixer.replaceTextRange(
+								[ quasi.start, quasi.end ],
+								replaceHyphenWithEnDash( quasi.value.raw )
+							)
+						);
+					}
+					return fixes;
+				}, [] );
```

Comment:
I noticed that the fix on template literals doesn't produce the expected result:

Input: ```__( `1-2` )```
Output: ```__( 1–2 )```
**Expected output:** ```__( `1–2` )```

However, looks like this issue can also be reproduced with the rule `i18n-ellipsis`:

Input: __( `Some text...` )
Output: __( Some text… )
**Expected output:** __( `Some text…` )

Not sure if you based the implementation on that rule but at least these two produce the same issue.

<hr>

Probably this issue is related to the text range provided in line 32.

Reply:
I tried adjusting the fixer here and nothing seemed to work quite right. I also noticed that the ellipsis fixer also has issues with template literals. 
I opted to just remove the fixer for template literals, the rule will still catch them but for now the author will have to fix them manually.

Reply:
> Ah correct, I'll go ahead and remove the fixer here too.

Just in case it was overlooked, I noticed in the merge commit https://github.com/WordPress/gutenberg/commit/1a8971ba51d1f69b718bf2f77261184b43950217#diff-86670005a144398f0d3c02df4d5e838e32707224a8495b13c97fce05b086feef that the fixer wasn't removed.

> TBH the fixers are "nice to haves" so don't want to spend too much time on them.

I agree, the important part is the rule itself, I'd consider fixers as optional/nice-to-have 👍 .